lib/testHash.ml

```diff
@@ -48,7 +48,8 @@ let digest_init debug init =
         | 0 ->
             if
               ParsedConstant.compare v1 v2 <> 0 &&
-              compare t1 t2 <> 0
+              compare t1 t2 <> 0 &&
+              (not (is_address t1 <> is_address t2));
```

Comment:
```suggestion
              is_address t1 = is_address t2
```
Incidentally, the final semicolon is accepted  and immediately forgotten by the `ocaml` parser. That is expression `e;` simply is `e`, except in sequence `e1; e2`. Something I have always found very bizarre.

Reply:
yeah this annoys me more than it should - it makes parser construction much harder than it needs to be. Will fix